### PR TITLE
image

### PR DESCRIPTION
**Pull request by gherghelalessandro**

### CODE REVIEW

File: source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Application/Services/UserService.cs [hunk 1]
```diff
@@ -52,10 +64,11 @@ public async Task Register(RegisterRequest Register)
         }
         public async Task<User> GetUser(Guid Id)
         {
+            string SasKey = "?sv=2021-06-08&ss=bfqt&srt=sco&sp=rwdlacupiytfx&se=2022-08-16T17:30:43Z&st=2022-07-20T09:30:43Z&sip=0.0.0.0-255.255.255.255&spr=https&sig=RReu5GnC4EjJqvE63A00A3iK6gLCOJp9Mk%2F6eXHbeQM%3D";
```
Comment (HincuAndreiAssist) [inline source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Application/Services/UserService.cs:67]: You should get the SasKey from appsettings

**Review by HincuAndreiAssist — CHANGES_REQUESTED** [review]

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 2]
```diff
@@ -35,37 +35,37 @@ public async Task<IActionResult> Authenticate(LoginRequest Login)
             }
             catch(AppException ex)
             {
-                return new BadRequestObjectResult(ex.Message);
+                return new BadRequestObjectResult("{'message':'user not found'}");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:38]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 3]
```diff
@@ -35,37 +35,37 @@ public async Task<IActionResult> Authenticate(LoginRequest Login)
...
         }
         
         [HttpPost("Register"),AllowAnonymus]
-        public async Task<IActionResult> Register(RegisterRequest Register)
+        public async Task<IActionResult> Register(string email,string password)
         {
             try
             {
-                await UserService.Register(Register);
+                await UserService.Register(email,password);
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:47]: you should use a DTO

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 4]
```diff
@@ -35,37 +35,37 @@ public async Task<IActionResult> Authenticate(LoginRequest Login)
...
 
-                return new OkObjectResult("ok");
+                return new OkObjectResult("{'message':'ok'};");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:49]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 5]
```diff
@@ -35,37 +35,37 @@ public async Task<IActionResult> Authenticate(LoginRequest Login)
...
             }
             catch (AppException ex)
             {
-                return new BadRequestObjectResult("An error has occured");
+                return new BadRequestObjectResult("{'message':'user already exist'}");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:53]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 6]
```diff
@@ -35,37 +35,37 @@ public async Task<IActionResult> Authenticate(LoginRequest Login)
...
-        [HttpPost("Reset/Password")]
+        [HttpPost("Reset/Password"),AllowAnonymus]
         public async Task<IActionResult> ResetPassword(string email)
         {
             try
             {
                 await UserService.ResetPassword(email);
 
-                return new OkObjectResult("ok");
+                return new OkObjectResult("{ 'message':'ok' };");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:64]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 7]
```diff
@@ -35,37 +35,37 @@ public async Task<IActionResult> Authenticate(LoginRequest Login)
...
             }
             catch(AppException ex)
             {
-                return new BadRequestObjectResult("An error has occured");
+                return new BadRequestObjectResult("{'message':'mail not found'}");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:68]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 8]
```diff
@@ -76,14 +76,14 @@ public async Task<IActionResult> Update(UpdateRequest Update,Guid id)
             {
                 await UserService.UpdateUser(Update,id);
 
-                return new OkObjectResult("ok");
+                return new OkObjectResult("{ 'message':'ok' }");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:79]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 9]
```diff
@@ -76,14 +76,14 @@ public async Task<IActionResult> Update(UpdateRequest Update,Guid id)
...
             }
             catch (AppException ex)
             {
-                return new NotFoundObjectResult("User not found");
+                return new NotFoundObjectResult("{'message':'user not found'}");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:83]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 10]
```diff
@@ -95,7 +95,7 @@ public async Task<IActionResult> GetUser(Guid id)
             }
             catch(AppException ex)
             {
-                return new NotFoundObjectResult("User not found");
+                return new NotFoundObjectResult("{'message':'User not found'}");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:98]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 11]
```diff
@@ -109,11 +109,11 @@ public async Task<IActionResult> GetUserEmail(string Email)
             }
             catch(AppException ex)
             {
-                return new NotFoundObjectResult("User not found");
+                return new NotFoundObjectResult("{'message':'user not found'}");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:112]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 12]
```diff
@@ -123,7 +123,7 @@ public async Task<IActionResult> GetAll()
             }
             catch(AppException ex)
             {
-                return new NotFoundObjectResult("Database is empty");
+                return new NotFoundObjectResult("{'message':'Empty'}");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:126]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 13]
```diff
@@ -134,12 +134,13 @@ public async Task<IActionResult> Delete(Guid id)
             {
                 await UserService.DeleteUser(id);
 
-                return new OkObjectResult("ok");
+                return new OkObjectResult("{ 'message':'ok' }");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:137]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 14]
```diff
@@ -134,12 +134,13 @@ public async Task<IActionResult> Delete(Guid id)
...
             }
             catch(AppException ex)
             {
-                return new NotFoundObjectResult("User not found");
+                return new NotFoundObjectResult("{'message':'user not found'}");
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:141]: you should return a DTO, not hardcode the message

File: source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Application/Interfaces/IUserService.cs [hunk 15]
```diff
@@ -1,12 +1,13 @@
...
 using Assist.July._2022.BE2.Domain.Entities;
+using Microsoft.AspNetCore.Http;
 
 namespace Assist.July._2022.BE2.Application.Interfaces
 {
     public interface IUserService
     {
         Task<User> Login(LoginRequest Login);
-        Task Register(RegisterRequest Register);
+        Task Register(string Email,string Password);
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Application/Interfaces/IUserService.cs:10]: space between properties, also the properties should be camelCase, not PascalCase

File: source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Application/Interfaces/IUserService.cs [hunk 16]
```diff
@@ -1,12 +1,13 @@
...
         Task<User> GetUser(Guid Id);
         Task<User> GetUserEmail(string Email);
         Task ResetPassword(string Email);
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Application/Interfaces/IUserService.cs:13]: same comment as before

File: source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Application/Interfaces/IUserService.cs [hunk 17]
```diff
@@ -1,12 +1,13 @@
...
 
 namespace Assist.July._2022.BE2.Application.Interfaces
 {
     public interface IUserService
     {
         Task<User> Login(LoginRequest Login);
-        Task Register(RegisterRequest Register);
+        Task Register(string Email,string Password);
         Task<User> GetUser(Guid Id);
         Task<User> GetUserEmail(string Email);
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Application/Interfaces/IUserService.cs:12]: camelCase

**Review by SebastianFulgaAssist — APPROVED** [review]

**Pull request merged**